src/topology/continuous_function/bounded.lean

```diff
@@ -1030,4 +1032,64 @@ module over the algebra of bounded continuous functions from `α` to `𝕜`. -/
 
 end normed_algebra
 
+section cstar_ring
+/-!
+### Star structure
+
+In this section, if `β` is a C⋆-ring, then the space of bounded
+continuous functions from `α` to `β` inherits a C⋆-ring structure also,
+by using the star operation pointwise. Moreover, if `𝕜` is normed field
+and a ⋆-ring, over which `β` is a normed algebra and a star module, then
+the space of bounded continuous functions from `α` to `β` is a star module
+over `𝕜`. In other words, `α →ᵇ β` is a C⋆-algebra (completeness is
+guaranteed when `β` is complete). -/
+
+variables {𝕜 : Type*} [normed_field 𝕜] [star_ring 𝕜]
+variables [topological_space α] [normed_ring β] [star_ring β] [cstar_ring β]
+variables [normed_algebra 𝕜 β] [star_module 𝕜 β]
+
+/-
+Here we can get away with `[normed_star_monoid β]` instead of `[cstar_ring β]`.
+Then change would be to replace :
+  `cstar_ring.to_normed_star_monoid.norm_star`
+with :
+  `(@normed_star_monoid.norm_star _ _ _ _)`
+But since we are mainly focused on showing that `α →ᵇ β` is a `cstar_ring`, we
+just do the slightly less general thing in favor of simplifying `variables`
+declarations.
+-/
+instance : star_ring (α →ᵇ β) :=
+{ star := λ f,
+    have iso : isometry (star : β → β), from
+      star_add_equiv.to_add_monoid_hom.isometry_of_norm cstar_ring.to_normed_star_monoid.norm_star,
+    comp star iso.lipschitz f,
```

Comment:
I propose adding the following construction to `analysis.normed_space.star` (useful in any case, as the next-best-thing to the non-existent `normed_group` version of `linear_isometry_equiv`)
```lean
def star_normed_group_hom {E : Type*} [normed_group E] [star_add_monoid E] [normed_star_monoid E] :
  normed_group_hom E E :=
{ bound' := ⟨1, λ v, by simpa using norm_star.le⟩,
  .. star_add_equiv }
```

Then this line becomes
```suggestion
{ star := λ f, f.comp star star_normed_group_hom.lipschitz,
```

Reply:
Your proof for `bound'` works with just `simp`, but the term was short enough to construct by hand that I just did that. If you would prefer, I can change it back to `by simp`